test/unit/utils/scheduler.spec.js

```diff
@@ -154,4 +154,16 @@ describe('createScheduler tasks execution', () => {
     const promise = Promise.race([originalTask, replacedTask])
     return expect(promiseTimeout(200, promise)).resolves.toBe(undefined)
   })
+
+  test('handles runImmediately correctly', () => {
+    const scheduler = createScheduler()
+
+    const promise = new Promise(resolve => {
+      scheduler.addTask({ task: () => resolve(true), baseDelay: 10, runImmediately: true })
+    })
+
+    scheduler.removeAllTasks()
+
+    return expect(promiseTimeout(100, promise)).resolves.toBe(true)
```

Comment:
Shouldn't this be set to `5ms` in order to verify that the task is run and resolves before the `10ms` base delay?

Reply:
i guess 5ms wouldn't hurt, but this test assumes `removeAllTasks` actually works. Otherwise thing thing should never resolve and 100ms is just to not hang the test process. So tldr i think using 5ms is even better